Reorganize Get Started (#362)
pingcap/docs-tidb-operator: en/get-started.md

+ On public cloud:
    - [Deploy TiDB on AWS EKS](deploy-on-aws-eks.md)
    - [Deploy TiDB on GCP GKE (beta)](deploy-on-gcp-gke.md)
    - [Deploy TiDB on Alibaba Cloud ACK](deploy-on-alibaba-cloud.md)

+ In an existing Kubernetes cluster:
    1. Familiarize yourself with [Prerequisites for TiDB in Kubernetes](prerequisites.md)
    2. Configure the local PV for your Kubernetes cluster to achieve low latency of local storage for TiKV according to [Local PV Configuration](configure-storage-class.md#local-pv-configuration)
    3. Install TiDB Operator in a Kubernetes cluster according to [Deploy TiDB Operator in Kubernetes](deploy-tidb-operator.md)
    4. Deploy your TiDB cluster according to [Deploy TiDB in General Kubernetes](deploy-on-general-kubernetes.md)

## Create a Kubernetes Cluster

This section covers 3 different ways to create a simple Kubernetes cluster that can be used to test TiDB Cluster running under TiDB Operator. Choose whichever best matches your environment or experience level.

- [Using kind](#create-a-kubernetes-cluster-using-kind) (Kubernetes in Docker)
- [Using minikube](#create-a-kubernetes-cluster-using-minikube) (Kubernetes running locally in a VM)

> - [Using the Google Cloud Shell](deploy-tidb-from-kubernetes-gke.md) (Kubernetes running in the Google Kubernetes Engine in Google Cloud Platform)

Reviewer: No procedure for this way in the following sections.

Author: Correct, you have to click that link, which will take you to the page with those instructions. I had a lot of trouble figuring out how to properly organize that, so I kind of gave up. If you have a suggestion, please let me know!

Author: Maybe better would be to just add a link directly to the Cloud Shell *here* and a little explanation for what is happening.

### Create a Kubernetes Cluster Using kind

> This tutorial shows how to deploy [TiDB Operator](https://github.com/pingcap/tidb-operator) and a TiDB cluster in Kubernetes on your laptop (Linux or macOS) using .sigs.k8s.io/).

Reviewer: ```suggestion
This section shows how to deploy a Kubernetes cluster using kind.
```

The following is an example of using `kind` v0.8.1:

{{< copyable "shell-regular" >}}

```shell
kind create cluster
```

Expected output:
```
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.18.2) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Thanks for using kind! 😊
```

Check whether the cluster is successfully created:

{{< copyable "shell-regular" >}}

```shell
kubectl cluster-info
```

Expected output:
```
Kubernetes master is running at https://127.0.0.1:51026
KubeDNS is running at https://127.0.0.1:51026/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

You're now ready to [Deploy TiDB Operator](#deploy-tidb-operator)!

To destroy the Kubernetes cluster, run the following command:

{{< copyable "shell-regular" >}}

``` shell
kind delete cluster
```

### Create a Kubernetes Cluster Using minikube

> This document describes how to deploy a TiDB cluster in the [minikube](https://kubernetes.io/docs/setup/minikube/) cluster.

Reviewer: ```suggestion
This section describes how to deploy a Kubernetes cluster using minikube.
```

        NOTES:
        1. Make sure tidb-operator components are running
           kubectl get pods --namespace tidb-admin -l app.kubernetes.io/instance=tidb-operator
        2. Install CRD
           kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/manifests/crd.yaml
           kubectl get customresourcedefinitions
        3. Modify tidb-cluster/values.yaml and create a TiDB cluster by installing tidb-cluster charts
           helm install tidb-cluster
        ```

        Confirm that the TiDB Operator components are running with this command:

        {{< copyable "shell-regular" >}}

        ```shell
        kubectl get pods --namespace tidb-admin -l app.kubernetes.io/instance=tidb-operator
        ```

        Expected output:
        ```
        NAME                                       READY   STATUS    RESTARTS   AGE
        tidb-controller-manager-6d8d5c6d64-b8lv4   1/1     Running   0          2m22s
        tidb-scheduler-644d59b46f-4f6sb            2/2     Running   0          2m22s
        ```

        As soon as all pods are in "Running" state, proceed to the next step.

> 2. Install TiDB Operator CRDs:

Reviewer: This should be done before the TiDB Operator installation.

Author: OK

Author: I will fix this in a separate commit.

    STATUS_ADDRESS: 0.0.0.0:20180
           VERSION: 4.0.0
          GIT_HASH: 198a2cea01734ce8f46d55a29708f123f9133944
        START_TIME: 2020-05-28T22:48:21Z
            UPTIME: 5m11.459102648s
    *************************** 6. row ***************************
              TYPE: tikv
          INSTANCE: basic-tikv-2.basic-tikv-peer.tidb-cluster.svc:20160
    STATUS_ADDRESS: 0.0.0.0:20180
           VERSION: 4.0.0
          GIT_HASH: 198a2cea01734ce8f46d55a29708f123f9133944
        START_TIME: 2020-05-28T22:47:17Z
            UPTIME: 6m15.459103932s
    6 rows in set (0.01 sec)
    ```

4. Load Grafana dashboard

    As done above for port 4000 to the TiDB service, also forward the port for Grafana so we can load the monitoring dashboard:

    {{< copyable "shell-regular" >}}

    ``` shell
    kubectl port-forward -n tidb-cluster svc/basic-grafana 3000 > pf3000.out &
    ```

    The dashboard will be accessible at <http://localhost:3000> on the host where you've run `kubectl`. Note that if you're running `kubectl` in a Docker container or on a remote host, you may not be able to load this URL and additional networking beyond the scope of this document may be necessary.

>     The default username and password in Grafana are both "admin".

Reviewer: We need to clear the `port-forward` in steps 2 & 4.

Author: What do you mean by "clear"?

Reviewer: The port-forward process is started in the background, after we're done with the demo access, we'd better stop the port-forward process, otherwise, it will always run and when you use this command to port-forward again or for another cluster, it will fail.

Author: Oh, OK. In my experience, it often exits on its own, but this is good advice.

    kubectl delete namespace tidb-cluster
    ```

5. Stop `kubectl` port forwarding

    If you still have running `kubectl` processes that are forwarding ports, end them.

    {{< copyable "shell-regular" >}}

    ```shell

>     pgrep -lfa kubectl

Reviewer: Do we need to mention how users can end the processes? The command here just lists the processes.
